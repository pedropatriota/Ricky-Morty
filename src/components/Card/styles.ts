import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
