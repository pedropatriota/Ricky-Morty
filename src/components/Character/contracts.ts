export interface ICharacterProps {
	id: number;
	name: string;
	species: string;
	status: string;
	gender: string;
	image: string;
}
