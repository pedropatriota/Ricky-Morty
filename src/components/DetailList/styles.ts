import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin-bottom: 0.75rem;
`;
