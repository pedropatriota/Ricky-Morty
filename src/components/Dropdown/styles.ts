import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
