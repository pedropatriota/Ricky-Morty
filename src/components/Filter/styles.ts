import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	align-items: flex-end;
	justify-content: space-between;
	flex-wrap: wrap;
	gap: 0.8rem;

	@media (max-width: 500px) {
		flex-wrap: wrap;
		gap: 1rem;
	}
`;
