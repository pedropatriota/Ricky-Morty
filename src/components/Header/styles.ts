import styled from 'styled-components';

export const Header = styled.header`
	padding: 0px min(5vw, 5rem);
	background-color: ${({ theme }) => theme.colors.bg};
	box-shadow: 0px 10px 15px -3px rgba(0, 0, 0, 0.1);
	margin-bottom: 4rem;
`;

export const Container = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	flex-wrap: wrap;

	img {
		width: 100%;
		height: auto;
		max-width: 10rem;
	}
`;
