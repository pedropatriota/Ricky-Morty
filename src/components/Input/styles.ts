import styled from 'styled-components';

export const Input = styled.input`
  width: min(90vw, 30rem);
  border: ${(props) => props.theme.colors.border};
  color: ${(props) => props.theme.colors.text};
  height: 3.375rem;
  border-radius: 3px;
  padding: 0 2rem;
  font-size:0.94rem;
  outline: none;
  background: ${(props) => props.theme.colors.el}
`;
