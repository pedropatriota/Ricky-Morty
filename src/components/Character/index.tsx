import { Link } from 'react-router-dom';
import Card from '../Card';
import type { ICharacterProps } from './contracts';

const Character = ({ id, name, species, status, gender, image }: ICharacterProps) => {
	return (
		<Link
			to={`/${id}`}
			style={{ textDecoration: 'none', width: 'fit-content', display: 'block' }}
		>
			<Card
				name={name}
				image={image}
				species={species}
				status={status}
				gender={gender}
			/>
		</Link>
	);
};

export default Character;
