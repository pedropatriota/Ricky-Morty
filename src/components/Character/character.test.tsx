import { screen, render, fireEvent, waitFor } from '@testing-library/react';
import Character from '.';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { light } from '../../styles/themes';

const setup = () => {
	const props = {
		id: 1,
		gender: 'Male',
		image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
		name: 'Rick Sanchez',
		species: 'Human',
		status: 'Alive'
	};

	const utils = render(
		<ThemeProvider theme={light}>
			<MemoryRouter>
				<Character {...props} />
			</MemoryRouter>
		</ThemeProvider>
	);

	const link = screen.getByRole('link', {
		name: /rick sanchez status: alive species: human gender: male/i
	}) as HTMLAnchorElement;

	return {
		link,
		props,
		...utils
	};
};

describe('Character', () => {
	it('should render correctly', () => {
		const { link } = setup();

		expect(link).toBeInTheDocument();
	});
	it('should go to detail page when clicked', async () => {
		const { link, props } = setup();

		fireEvent.click(link);

		await waitFor(() => {
			expect(link.href).toContain(`/${props.id}`);
		});
	});
	it('should match the snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});
});
