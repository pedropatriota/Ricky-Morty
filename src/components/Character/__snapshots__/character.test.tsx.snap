// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should match the snapshot 1`] = `
<div>
  <a
    href="/1"
    style="text-decoration: none; display: block;"
  >
    <div
      class="sc-aXZVg bNflSf"
      data-testid="card"
    >
      <div
        class="sc-gEvEer cvOKXx"
        data-testid="card-image"
      />
      <div
        class="sc-eqUAAy jAJozm"
      >
        <h2>
          Rick Sanchez
        </h2>
        <p>
          <strong>
            Status
            : 
          </strong>
           
          Alive
        </p>
        <p>
          <strong>
            Species
            : 
          </strong>
           
          Human
        </p>
        <p>
          <strong>
            Gender
            : 
          </strong>
           
          Male
        </p>
      </div>
    </div>
  </a>
</div>
`;
